spring-boot-project/spring-boot/src/test/java/org/springframework/boot/context/properties/ConfigurationPropertiesTests.java

```diff
@@ -1270,6 +1271,25 @@ void loadWhenBindingToJavaBeanWithConversionToCustomListImplementation() {
 		assertThat(this.context.getBean(SetterBoundCustomListProperties.class).getValues()).containsExactly("a", "b");
 	}
 
+	@Test
+	void autowiredConstructorShouldInjectBeanInsteadOfBindProperties() {
+		load(ServicePropertiesConfiguration.class, "primary.service.port=1234", "foo.service.host=127.0.0.1");
+		PrimaryServiceProperties primaryServiceProperties = this.context.getBean(PrimaryServiceProperties.class);
+		assertThat(primaryServiceProperties.getHost()).isEqualTo("localhost");
+		assertThat(primaryServiceProperties.getPort()).isEqualTo(1234);
+		FooServiceProperties fooServiceProperties = this.context.getBean(FooServiceProperties.class);
+		assertThat(fooServiceProperties.getHost()).isEqualTo("127.0.0.1");
+		assertThat(fooServiceProperties.getPort()).isEqualTo(1234);
+	}
+
+	@Test
+	void privateConstructorShouldInjectBeanInsteadOfBindProperties() {
+		load(ServicePropertiesConfiguration.class, "primary.service.port=1234", "bar.service.host=127.0.0.1");
+		BarServiceProperties barServiceProperties = this.context.getBean(BarServiceProperties.class);
+		assertThat(barServiceProperties.getHost()).isEqualTo("127.0.0.1");
+		assertThat(barServiceProperties.getPort()).isEqualTo(1234);
+	}
+
```

Comment:
This is similar to this test:

https://github.com/spring-projects/spring-boot/blob/2bfd784d7baef889fab89ab406f9fcb762b81c6d/spring-boot-project/spring-boot/src/test/java/org/springframework/boot/context/properties/bind/DefaultBindConstructorProviderTests.java#L95-L100

If we want to cover the non-member case as well, I think it should be done in `DefaultBindConstructorProviderTests` rather than here as it's `DefaultBindConstructorProvider` that contains the logic for ignoring private constructors:

https://github.com/spring-projects/spring-boot/blob/2bfd784d7baef889fab89ab406f9fcb762b81c6d/spring-boot-project/spring-boot/src/main/java/org/springframework/boot/context/properties/bind/DefaultBindConstructorProvider.java#L188

Reply:
I’ll add unit test tomorrow.

Reply:
done.